From 6a82d9a966f9e3d5da928be8202ea18de1fb2d65 Mon Sep 17 00:00:00 2001
From: Alejandro Martinez Ruiz <alex@nowcomputing.com>
Date: Mon, 10 Jun 2013 14:21:53 +0200
Subject: [PATCH] f2fs: add ->check_acl() test for IPERM_FLAG_RCU

This reverts the effect of upstream commit 9c2c703 ->permission()
sanitizing: pass MAY_NOT_BLOCK to ->check_acl().

Signed-off-by: Alejandro Martinez Ruiz <alex@nowcomputing.com>
---
 fs/f2fs/acl.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/fs/f2fs/acl.c b/fs/f2fs/acl.c
index 0ae210e..d5cf319 100644
--- a/fs/f2fs/acl.c
+++ b/fs/f2fs/acl.c
@@ -193,7 +193,7 @@ int f2fs_check_acl(struct inode *inode, int mask, unsigned int flags)
 {
 	int error = -EAGAIN;
 
-	if (mask & MAY_NOT_BLOCK) {
+	if (flags & IPERM_FLAG_RCU) {
 		if (!negative_cached_acl(inode, ACL_TYPE_ACCESS))
 			error = -ECHILD;
 	} else {
-- 
1.8.2.1.422.gdcd8c09

